cas-server-core/src/main/java/org/jasig/cas/util/CipherExecutor.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.jasig.cas.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Responsible to define operation that deal with encryption, signing
+ * and verification of a value.
+ *
+ * @author Misagh Moayyed
+ * @since 4.1
+ */
+public interface CipherExecutor {
+
+    /**
+     * Encrypt the value. Implementations may
+     * choose to also sign the final value.
+     * @param value the value
+     * @return the encrypted value or null
+     */
+    String encode(String value);
+
+    /**
+     * Decode the value. Signatures may also be verified.
+     * @param value encrypted value
+     * @return the decoded value. 
+     */
+    String decode(String value);
+
+    /**
+     * No-Op cipher executor that does nothing for encryption/decryption.
+     * @author Misagh Moayyed
+     * @since 4.1
+     */
+    public static final class NoOpCipherExecutor implements CipherExecutor {
```

Comment:
Perhaps, but I didnt want to add yet another java class file to host 4 lines of code. We could move it out though. It's just a matter of organization.